src/platform/datapath_kqueue.c

```diff
@@ -1211,7 +1238,10 @@ CxPlatSocketContextRecvComplete(
                 FoundTOS = TRUE;
             }
         } else if (CMsg->cmsg_level == IPPROTO_IP) {
-            if (CMsg->cmsg_type == IP_PKTINFO) {
+#if defined(DSTADDR_SOCKOPT)
+            if (CMsg->cmsg_type == DSTADDR_SOCKOPT) {
+#else
+#endif
```

Comment:
You have to have something here or the `else if` below won't work.

Reply:
Well I noticed
```
// For all *NIX
#if defined(IP_RECVDSTADDR)
#define DSTADDR_SOCKOPT IP_RECVDSTADDR
#define DSTADDR_DATASIZE (CMSG_SPACE(sizeof(struct in_addr)))
#define dstaddr(x) (CMSG_DATA(x))

#elif defined(IP_PKTINFO)
#define DSTADDR_SOCKOPT IP_PKTINFO
#define DSTADDR_DATASIZE (CMSG_SPACE(sizeof(struct in_pktinfo)))
#define dstaddr(x) (&(((struct in_pktinfo *)(CMSG_DATA(x)))->ipi_addr))

#endif

```

should be moved under #include <*.h>.

Reply:
Could you try the latest update?